cache/lru_cache.cc

```diff
@@ -513,21 +516,22 @@ Cache::Handle* LRUCacheShard::Lookup(
     e = table_.Lookup(key, hash);
     if (e != nullptr) {
       assert(e->InCache());
-      if (!e->HasRefs()) {
-        // The entry is in LRU since it's in hash and has no external
-        // references.
-        LRU_Remove(e);
-      }
-      e->Ref();
-      e->SetHit();
-
-      // For a dummy handle, if it was retrieved from secondary cache,
-      // it may still exist in secondary cache.
-      // If the handle exists in secondary cache, the value should be
-      // erased from sec cache and be inserted into primary cache.
-      if (!e->value && secondary_cache_ &&
-          secondary_cache_->SupportForceErase()) {
+      if (e->value == kDummyValueMarker) {
+        // For a dummy handle, if it was retrieved from secondary cache,
+        // it may still exist in secondary cache.
+        // If the handle exists in secondary cache, the value should be
+        // erased from sec cache and be inserted into primary cache.
         found_dummy_entry = true;
+        // Let the dummy entry be overwritten
+        e = nullptr;
```

Comment:
This may cause an ASAN issue.

Reply:
`table_.Lookup(key, hash)` does not acquire a reference